### PR TITLE
Revert "Improve Iceberg deletes when an entire file can be removed"

### PR DESCRIPTION
**Pull request by alexjo2144**
## Description

This reverts commit 9ddaa60df18f74247d299cf0ce4c81d9383cfb9d.

Including both a RowDelta and a DeleteFiles commit in the same Iceberg transaction
results in the table history being cleared. Instead only write RowDeltas,
even in the case where a file can be fully removed.

> Is this change a fix, improvement, new feature, refactoring, or other?

Fix

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

Iceberg connector

> How would you describe this change to a non-technical end user or system administrator?

Ensures Trino does not clear Iceberg table history when it's not expected.

## Related issues, pull requests, and links

Quick fix for: https://github.com/trinodb/trino/issues/12843

## Documentation

(x) No documentation is needed.
( ) Sufficient documentation is included in this PR.
( ) Documentation PR is available with #prnumber.
( ) Documentation issue #issuenumber is filed, and can be handled later.

## Release notes

( ) No release notes entries required.
(x) Release notes entries required with the following suggested text:

```markdown
# Iceberg
* Fix delete transactions to ensure table history remain intact.
```

Fixes https://github.com/trinodb/trino/issues/12843

### CODE REVIEW

**Comment by alexjo2144** [comment]
@findepi added a regression test

**Comment by findepi** [comment]
@alexjo2144 please see build failures

**Comment by alexjo2144** [comment]
The changes introduced by Merge made this more complex than a simple revert, but should be passing now.

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java [hunk 1]
```diff
@@ -5268,6 +5268,22 @@ public void testReadFromVersionedTableWithExpiredHistory()
...
+    {
+        String tableName = "test_delete_retains_table_history_" + randomTableSuffix();
+        assertUpdate("CREATE TABLE " + tableName + "(c1 INT, c2 INT)");
+        assertUpdate("INSERT INTO " + tableName + " VALUES (1, 1), (2, 2), (3, 3)", 3);
+        assertUpdate("INSERT INTO " + tableName + " VALUES (3, 3), (4, 4), (5, 5)", 3);
+        List<Long> snapshots = getTableHistory(tableName);
+
+        assertUpdate("DELETE FROM " + tableName + " WHERE c1 < 4", 4);
+        List<Long> snapshotsAfterDelete = getTableHistory(tableName);
+        assertEquals(snapshotsAfterDelete.size(), snapshots.size() + 1);
```
Comment (findepi) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java:5282]: I'd drop this, relax, or add a comment.

In fact we accept situation where deletion results in two snapshots, right?

**Review by findepi — APPROVED** [review]

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java (hunk 1 above)
Comment (alexjo2144) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java]: Sure. Switched it to a greater than check

**Review by findepi — APPROVED** [review]

**Comment by findepi** [comment]
> Fixes #12843

FYI, added  this in the description.

**Pull request merged**